app/default/account_routes.py

```diff
@@ -1,67 +1,109 @@
-
 import requests
+from app.default.data import get_all_funds
+from app.default.data import get_all_rounds_for_fund
 from app.default.data import get_applications_for_account
-from app.default.data import get_round_data_fail_gracefully
+from app.models.application_summary import ApplicationSummary
+from config import Config
+from flask import Blueprint
+from flask import current_app
+from flask import g
+from flask import redirect
+from flask import render_template
 from flask import request
 from flask import url_for
 from fsd_utils.authentication.decorators import login_required
 from fsd_utils.locale_selector.get_lang import get_lang
-from app.models.application_summary import ApplicationSummary
-from flask import current_app
-from config import Config
-from flask import Blueprint, g, redirect, render_template
-from app.default.routes import current_datetime_after_given, current_datetime_before_given
+from fsd_utils.simple_utils.date_utils import (
+    current_datetime_after_given_iso_string,
+)
+from fsd_utils.simple_utils.date_utils import (
+    current_datetime_before_given_iso_string,
+)
 
 
 account_bp = Blueprint("account_routes", __name__, template_folder="templates")
 
+
+def build_application_data_for_display(applications: list[ApplicationSummary]):
+
+    application_data_for_display = {"funds": []}
+
+    count_of_applications_for_visible_rounds = 0
+    all_funds = get_all_funds()
+    if all_funds:
```

Comment:
Could get rid of a block of identation/make more readable if we make use of the "return early pattern" here

```suggestion
    application_data_for_display = {
        "funds": [],
        "total_applications_to_display": 0,
    }

    all_funds = get_all_funds()
    if not all_funds: return count_of_applications_for_visible_rounds 
    
    count_of_applications_for_visible_rounds = 0
    for fund in all_funds:
        ...
```